doc/mgmt/SONiC_OC_MGMT_Interface_HLD.md

```diff
@@ -0,0 +1,261 @@
+# Feature Name
+Openconfig support for Physical and Management interfaces via openconfig-interfaces.yang.
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 09/09/2019  |   Ravi Vasanthm     | Initial version                   |
+
+# About this Manual
+This document provides general information about openconfig support for Physical and Management interfaces handling in SONIC.
+
+# Scope
+This document describes the high level design of openconfig support for Physical and Management interfaces handling feature. Call out any related design that is not covered by this document
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| MGMT Intf                     | Management Interface                     |
+
+# 1 Feature Overview
+This feature will provide config set/get and status get support for Physical and Management interfaces according to the openconfig-interfaces.yang data model via CLI, REST, and gNMI.
+
+https://github.com/project-arlo/sonic-mgmt-framework/blob/master/models/yang/openconfig-interfaces.yang
+
+## 1.1 Requirements
+
+
+### 1.1.1 Functional Requirements
+
+1. Provide CLI, REST and GNMI support for configuring and displaying physical and management interfaces attributes.
+2. Enhance existing implementation of interfaces yang to include physical and management interfaces handling.
+3. Enhance existing top level show commands for interfaces to include physical and management interfaces details too.
+
+
+### 1.1.2 Configuration and Management Requirements
+1. Provide CLI/GNMI/REST support for configuring Physical and Management interfaces attributes.
+2. Provide CLI/GNMI/REST support for show Physical and Management  interfaces attributes/parameters.
+
+
+### 1.1.3 Scalability Requirements
+N/A
+### 1.1.4 Warm Boot Requirements
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+Will be enhancing the management framework backend and transformer methods to add support for Physical and  Management interfaces Handling.
+
+
+### 1.2.2 Container
+All code changes will be done in management-framework container.
+
+### 1.2.3 SAI Overview
+ N/A
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+Manage/configure physical and management interfaces via GNMI, REST and CLI interfaces
+
+## 2.2 Functional Description
+Provide GNMI and REST support for get/set of Physical and Management interfaces attributes and CLI config and show commands to manage Management interface.
```

Comment:
"get/set of Physical and Management interfaces attributes and CLI config and show commands to manage Management interface." 
Just the Management interface ? Seems contradictory

Reply:
Addressed